File: Excel/900-999/936/936 Challenge.R
library(tidyverse)
library(readxl)

path <- "900-999/936/936 Extract Name ID.xlsx"
input <- read_excel(path, range = "A2:A11")
test <- read_excel(path, range = "C2:E19")

pattern <- "([A-Z]+)\\s+([A-Z]+)\\s+\\((\\d+)\\)"

result <- input |>
  drop_na() |>
  mutate(matches = str_extract_all(Data, pattern)) |>
  unnest(matches) |>
  mutate(
    Surname = str_match(matches, pattern)[, 2],
    `First Name` = str_match(matches, pattern)[, 3],
    ID = as.integer(str_match(matches, pattern)[, 4])
  ) |>
  select(Surname, `First Name`, ID)

all.equal(result, test)
# [1] TRUE

File: Excel/900-999/936/936 Challenge.py
import pandas as pd
import re

path = "900-999/936/936 Extract Name ID.xlsx"
input_df = pd.read_excel(path, usecols="A", skiprows=1, nrows=10, header=0)
test = pd.read_excel(path, usecols="C:E", skiprows=1, nrows=18, header=0)

pattern = r'([A-Z]+)\s+([A-Z]+)\s+\((\d+)\)'

records = []
for val in input_df["Data"].dropna():
    for m in re.finditer(pattern, val):
        records.append({"Surname": m.group(1), "First Name": m.group(2), "ID": int(m.group(3))})

result = pd.DataFrame(records).reset_index(drop=True)

print(result.equals(test))
# True
